### ISSUE TITLE
An UploadedFile - early instantiation problem

### ISSUE THREAD

**Issue by daydiff** [yiisoft/yii2#9027]
If we use a few (more than one) inheritors of the ``UploadedFile`` in a controller than a ``getInstance()`` method called on any of inheritors returns the instance of inheritor that was used first in current php session.

Here an example:
```php
class ProductController extends \yii\web\Controller {
    public function actionUpdate($id)
    {
        $productImage = \app\components\ProductImage::getInstance($this, 'prod_image'); //returns an instance of ProductImage as expected
        $vendorImage = \app\components\VendorImage::getInstance($this, 'vendor_image'); //returns an instance of ProductImage - expected an instance of VendorImage!
    }
}
```

The problem is that ``yii\web\UploadedFile`` loads all files and instantiate its when fisrt call of ``getInstance()`` occurs using late static binding. So, when next calls of ``getInstance()`` occurs we getting a wrong instance from ``$_files`` static property.

**Comment by mdmunir** [comment]
why you need to extend `yii\web\UploadedFile`?

**Comment by daydiff** [comment]
Why not? It's not about my business logic. Btw, I need to implement some custom naming and manipulation for uploaded files and I don't want to put it into main model. I suppose that inheritor of ``yii\web\UploadedFile`` the best solution to do that.

Anyway, if ``yii\web\UploadedFile`` isn't declared as a final class, it's ``getInstance()`` method must return an instance of a class on which we call that method. It's logical and expected behavior.

**Comment by klimov-paul** [comment]
Resolved by commit  2fe0a40
